### PR TITLE
LPS-171649

### PR DESCRIPTION
**Pull request by alinedoleron**
Ticket: https://issues.liferay.com/browse/LPS-171649

### CODE REVIEW

**Comment by alinedoleron** [comment]
ci:test:sf

File: modules/apps/object/object-js-components-web/src/main/resources/META-INF/resources/utils/string.ts [hunk 1]
```diff
@@ -12,6 +12,24 @@
  * details.
  */
 
+const defaultLanguageId = Liferay.ThemeDisplay.getDefaultLanguageId();
+
+type Label = {
+	[key: string]: string;
+};
```
Comment (danvb112) [inline modules/apps/object/object-js-components-web/src/main/resources/META-INF/resources/utils/string.ts:19]: Hey @alinedoleron, We don't need to create a new type for this, we can use an existing type in the Label which is: `LocalizedValue<string>`

File: modules/apps/object/object-js-components-web/src/main/resources/META-INF/resources/utils/string.ts [hunk 2]
```diff
@@ -12,6 +12,24 @@
...
+
+/**
+ * Get the label according to the locale
+ */
+
+export function getLocale(
```
Comment (danvb112) [inline modules/apps/object/object-js-components-web/src/main/resources/META-INF/resources/utils/string.ts:25]: hey @alinedoleron, I think we can rename this function to `getLocalizableLabel`, it makes more sense to me, what do you think?

File: modules/apps/object/object-js-components-web/src/main/resources/META-INF/resources/utils/string.ts [hunk 3]
```diff
@@ -12,6 +12,24 @@
...
+	labels: Label,
+	creationLanguageId: string,
```
Comment (danvb112) [inline modules/apps/object/object-js-components-web/src/main/resources/META-INF/resources/utils/string.ts:27]: hey @alinedoleron, to further specify the type of `creationLanguageId`, we can use an existing type which is: `Locale`

**Review by danvb112 — CHANGES_REQUESTED** [review]
Hey @alinedoleron, good job! I have some suggestions :smile:

File: modules/apps/object/object-js-components-web/src/main/resources/META-INF/resources/utils/string.ts (hunk 1 above)
Comment (alinedoleron) [reply modules/apps/object/object-js-components-web/src/main/resources/META-INF/resources/utils/string.ts:19]: Thanks for the suggestions, Dan!

**Comment by alinedoleron** [comment]
ci:test:sf

**Comment by alinedoleron** [comment]
ci:test:relevant

**Review by danvb112 — APPROVED** [review]
LGTM! :tada:

**Comment by danvb112** [comment]
ci:test:sf

**Review by danvb112 — APPROVED** [review]
LGTM! :tada:

@naraandrade do you have any other suggestions?

**Comment by alinedoleron** [comment]
ci:test:sf

**Comment by alinedoleron** [comment]
ci:test:relevant

**Comment by danvb112** [comment]
ci:test:relevant

**Comment by alinedoleron** [comment]
cc: @M-IgorFranca

**Comment by alinedoleron** [comment]
ci:test:sf

**Review by naraandrade — APPROVED** [review]

**Comment by naraandrade** [comment]
ci:test:relevant

**Comment by naraandrade** [comment]
ci:forward

File: modules/apps/object/object-js-components-web/src/main/resources/META-INF/resources/utils/string.ts [hunk 4]
```diff
@@ -12,6 +12,30 @@
...
 
+const defaultLanguageId = Liferay.ThemeDisplay.getDefaultLanguageId();
+
+/**
+ * Get the label according to the locale
+ */
+
+export function getLocalizableLabel(
+	labels: LocalizedValue<string> | undefined,
+	creationLanguageId: Locale,
```
Comment (naraandrade) [inline modules/apps/object/object-js-components-web/src/main/resources/META-INF/resources/utils/string.ts:23]: @alinedoleron could you sort the parameters alphabetically, please? Thanks :)

**Review by naraandrade — CHANGES_REQUESTED** [review]

File: modules/apps/object/object-js-components-web/src/main/resources/META-INF/resources/utils/string.ts (hunk 4 above)
Comment (alinedoleron) [reply modules/apps/object/object-js-components-web/src/main/resources/META-INF/resources/utils/string.ts:22]: Thanks, @naraandrade. I could only sort the non optional parameters. The build complains if a sort all the parameters: 

> "A required parameter cannot follow an optional parameter."

**Comment by alinedoleron** [comment]
ci:test:sf

**Comment by alinedoleron** [comment]
ci:test:sf

**Comment by alinedoleron** [comment]
ci:test:relevant

File: modules/apps/object/object-js-components-web/src/main/resources/META-INF/resources/utils/string.ts (hunk 4 above)
Comment (naraandrade) [reply modules/apps/object/object-js-components-web/src/main/resources/META-INF/resources/utils/string.ts:22]: Ok, Thank you!

**Comment by naraandrade** [comment]
@alinedoleron you can send this PR directly to Brian. No need to open a QA request since you're just creating a function and not applying it yet. Thanks :)

**Comment by alinedoleron** [comment]
PR manually sent to Brian: https://github.com/brianchandotcom/liferay-portal/pull/128497

**Pull request closed without merging**